api/v1alpha1/progressiverollout_types.go

```diff
@@ -90,6 +90,16 @@ func (in *ProgressiveRollout) NewStatusCondition(t string, s metav1.ConditionSta
 	}
 }
 
+// HasOwnerReference returns true if the ProgressiveRollout object has a reference to one of the owners
+func (in *ProgressiveRollout) HasOwnerReference(owners []metav1.OwnerReference) bool {
```

Comment:
I think I got slightly confused with the name of the function. It implies that the ProgressiveRollout has an attribute called OwnerRefrence and it would have one or not.
I'd rename it `IsReferredByOwners` or something like that?

Reply:
Yeah, done that as per @Smirl comment